miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppDisplay.kt

```diff
@@ -1,22 +1,35 @@
-package com.rakuten.tech.mobile.miniapp
+package com.rakuten.tech.mobile.miniapp.display
 
+import android.content.Context
 import android.view.View
+import androidx.lifecycle.Lifecycle
 import androidx.lifecycle.LifecycleObserver
+import com.rakuten.tech.mobile.miniapp.js.MiniAppMessageBridge
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
 
 /**
  * This represents the contract by which the host app can interact with the
  * display unit of the mini app.
- * This contract complies to Android's [LifecycleObserver] contract, and when made to observe
- * the lifecycle, it automatically clears up the view state and any services registered with.
  */
-interface MiniAppDisplay : LifecycleObserver {
+interface MiniAppDisplay {
+    val basePath: String
+    val appId: String
+    val miniAppMessageBridge: MiniAppMessageBridge
 
     /**
      * Provides the view associated with the mini app to the caller for showing the mini app.
+     * [LifecycleObserver] as optional. When made to observe the lifecycle,
+     * it automatically clears up the view state and any services registered with.
      * @return [View] as mini app's view with [LayoutParams] set to match
      * the parent's dimensions.
      */
-    fun getMiniAppView(): View
+    suspend fun getMiniAppView(context: Context, hostLifecycle: Lifecycle? = null): View =
+        withContext(Dispatchers.Main) {
+            val view = RealMiniAppDisplay(context, basePath, appId, miniAppMessageBridge)
```

Comment:
This will recreate the view every time that they call `getMiniAppView`. Could this cause problems? For example, wouldn't they expect it to always return the same instance if they call it multiple times? Though I can't think of any examples why they would want to call `getMiniAppView` more than once...

I know that I suggested passing the context at `getMiniAppView`, but now I'm wondering if it would be better to pass the context at `MiniApp#create`? It seems like it would solve some potential issues. What do you think @ts-nitish-a-jain @minh-rakuten ?

Reply:
I think I'll create another PR which targets at `MiniApp#create` so that we can compare. I agree that using `getMiniAppView` reveal issue such as `MiniAppDisplay` exposes the `basePath`.